src/plugins/formulas/formulas.js

```diff
@@ -20,6 +27,59 @@ export class Formulas extends BasePlugin {
     return PLUGIN_PRIORITY;
   }
 
+  /**
+   * Plugin settings.
+   *
+   * @private
+   */
+  #settings = this.hot.getSettings()[PLUGIN_KEY];
+
+  /**
+   * Flag used to bypass hooks in internal operations.
+   *
+   * @private
+   * @type {boolean}
+   */
+  #internal = false;
+
+  /**
+   * Flag needed to mark if Handsontable was initialized with no data.
+   * (Required to work around the fact, that Handsontable auto-generates sample data, when no data is provided).
+   *
+   * @type {boolean}
+   */
+  #emptyData = false;
```

Comment:
Perhaps rename to `hotWasInitializedWithEmptyData`? I know there's the jsdoc, but it could get confusing fast.